### PR TITLE
Reduce involvement of shared parental leave in pay leave for parents

### PR DESCRIPTION
**Pull request by cbaines**
This PR removes multiple outcomes from the Pay Leave For Parents flow around shared parental leave.

It removes 
* logic and tests for birth due dates prior to 2015-04-05
* questions related to shared parental leave
* several partials related to shared parental leave
* mother earnings question for employed and self-employed mothers with less than 26 weeks worked 

It adds
* copy for unemployed mother earnings and weeks worked questions
* copy for paternity leave and pay information for outcomes where they are NOT eligible

[Trello](https://trello.com/c/65jLhWvK/2019-3-end-of-jan-update-questions-and-outcomes-in-pay-leave-for-parents-to-remove-shared-parental-leave-information)

### CODE REVIEW

**Comment by theseanything** [comment]
@cbaines Whats the status of this PR?

**Comment by cbaines** [comment]
> @cbaines Whats the status of this PR?

As far as I'm aware, the changes are still with the content team/department.

I'm also not the best person to talk to about this any more, @thomasleese might have more up to date information.

**Comment by thomasleese** [comment]
> As far as I'm aware, the changes are still with the content team/department.
> 
> I'm also not the best person to talk to about this any more, @thomasleese might have more up to date information.

Yeah, we're still waiting on content.

**Review by MuriloDalRi — APPROVED** [review]

File: lib/smart_answer_flows/maternity-paternity-pay-leave.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 module SmartAnswer
-  class PayLeaveForParentsFlow < Flow
+  class MaternityPaternityPayLeaveFlow < Flow
     def define
       start_page_content_id "1f6b4ecc-ce2c-488a-b9c7-b78b3bba5598"
       flow_content_id "177cde4d-e52f-4629-bbbe-ec85a18ed944"
-      name "pay-leave-for-parents"
+      name "maternity-paternity-pay-leave"
```
Comment (thomasleese) [inline lib/smart_answer_flows/maternity-paternity-pay-leave.rb:6]: I think that since the content ID is the same, the Publishing API will automatically set up the redirect for this, which is nice.

**Review by thomasleese — APPROVED** [review]

File: lib/smart_answer_flows/maternity-paternity-pay-leave.rb (hunk 1 above)
Comment (edwardkerry) [reply lib/smart_answer_flows/maternity-paternity-pay-leave.rb:6]: This was my hope 🤞

File: lib/smart_answer_flows/pay-leave-for-parents.rb [hunk 2]
```diff
@@ -138,7 +138,42 @@ def define
         end
 
         next_node do
-          outcome :mother_earned_at_least_390
+          if %w[employee self-employed].include?(calculator.employment_status_of_mother) && calculator.mother_worked_at_least_26_weeks == "no"
```
Comment (ChrisBAshton) [inline lib/smart_answer_flows/pay-leave-for-parents.rb]: There are a lot of changes in this commit (0c41710492bdd7546d43614052ec37306456b534) - where did all this code come from? How can we be sure all these paths are tested?

Also, nitpick: the 1-line commit message is too long for GitHub's UI, could it be rephrased?

File: lib/smart_answer_flows/pay-leave-for-parents/outcomes/outcome_mat_leave_mat_pay.erb [hunk 3]
```diff
@@ -3,9 +3,9 @@
 
   <%= render partial: 'mat_pay', locals: { calculator: calculator } %>
 
-  <%= render partial: 'extra_help' %>
-
   <% if calculator.two_carers? %>
     <%= render partial: 'paternity_unavailable' %>
   <% end %>
+
+  <%= render partial: 'extra_help' %>
```
Comment (ChrisBAshton) [inline lib/smart_answer_flows/pay-leave-for-parents/outcomes/outcome_mat_leave_mat_pay.erb:10]: Could this commit be fixed up to the commit where this block is added?

```
  <% if calculator.two_carers? %>
    <%= render partial: 'paternity_unavailable' %>
  <% end %>
```

No point committing code in one commit and then moving it in the other 👍

**Review by ChrisBAshton — COMMENTED** [review]
I've reviewed from 0c41710492bdd7546d43614052ec37306456b534 onwards, on the assumption that Chris B's commits beforehand have already been reviewed.

A couple of comments below, but the commit in question is actually my only blocking comment - there seems to be a lot of change in the pay-leave-for-parents.rb file in that commit, and not necessarily the tests to back it up 🤔

File: lib/smart_answer_flows/pay-leave-for-parents/outcomes/outcome_mat_leave_mat_pay.erb (hunk 3 above)
Comment (edwardkerry) [reply lib/smart_answer_flows/pay-leave-for-parents/outcomes/outcome_mat_leave_mat_pay.erb:10]: I see what you mean...this is a symptom of changes being made months apart I think

File: lib/smart_answer_flows/pay-leave-for-parents.rb (hunk 2 above)
Comment (edwardkerry) [reply lib/smart_answer_flows/pay-leave-for-parents.rb]: This question used to always direct to `mother_earned_at_least_390` question - however for users under certain conditions who had answered "no" the input from `mother_earned_at_least_390` was irrelevant as their next step was already determined. So we are now able to direct users to their outcome as soon as possible.

This means this new code block and the existing `mother_earned_at_least_390`code is fairly similar.

https://github.com/alphagov/smart-answers/blob/0c41710492bdd7546d43614052ec37306456b534/lib/smart_answer_flows/pay-leave-for-parents.rb#L190

As the only change to the flow is missing out a question under certain conditions, the existing tests with those conditions failed. Removing the now missing question from the tests resolved the issue.

**Comment by edwardkerry** [comment]
> I've reviewed from [0c41710](https://github.com/alphagov/smart-answers/commit/0c41710492bdd7546d43614052ec37306456b534) onwards, on the assumption that Chris B's commits beforehand have already been reviewed.
> 
> A couple of comments below, but the commit in question is actually my only blocking comment - there seems to be a lot of change in the pay-leave-for-parents.rb file in that commit, and not necessarily the tests to back it up 🤔

Thanks Chris - in fact all these changes have previously been reviewed. I've hopefully answered the questions below, the changes are not as sizeable as they appear but hard to follow due to the very large conditionals we have in this flow.

File: lib/smart_answer_flows/pay-leave-for-parents.rb (hunk 2 above)
Comment (edwardkerry) [reply lib/smart_answer_flows/pay-leave-for-parents.rb]: I've also update the commit message with some more detail on this

Comment (ChrisBAshton) [reply lib/smart_answer_flows/pay-leave-for-parents.rb]: I've been having a think about this and have worked up a spike here: https://github.com/alphagov/smart-answers/pull/5254

Whilst the spike makes the code a bit more verbose (and does repeat some of the logic, as you suggested it might), I think it's a lot easier to follow and is probably a safer / more maintainable approach to working with Smart Answers from now on.

What do you think? If you like it, I'm tempted to write it up as an ADR for the repo. Either way, I'm happy to approve this PR, it looks good 👍

Comment (edwardkerry) [reply lib/smart_answer_flows/pay-leave-for-parents.rb]: I think this looks like a sensible approach - I'd suggest it's outside the scope of this PR but it's certainly worth refactoring the whole flow once this PR has been merged, and I'd be happy to take it on

**Review by ChrisBAshton — APPROVED** [review]

File: lib/smart_answer/calculators/maternity_paternity_pay_leave_calculator.rb [hunk 4]
```diff
@@ -121,7 +121,11 @@ def mother_earnings_employment?
 
     # Earnings and employment test
     def earnings_employment(earnings_employment, work_employment)
-      earnings_employment == "yes" && work_employment == "yes"
+      if employment_status_of_mother == "self-employed"
+        work_employment == "yes"
+      else
+        earnings_employment == "yes" && work_employment == "yes"
+      end
```
Comment (ChrisBAshton) [inline lib/smart_answer/calculators/maternity_paternity_pay_leave_calculator.rb:128]: I think this could be simplified to:

```ruby
return false unless work_employment == "yes"

employment_status_of_mother == "self-employed" || earnings_employment == "yes"
```

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍
